lib/meadow/ingest/work_creator.ex

```diff
@@ -23,21 +23,26 @@ defmodule Meadow.Ingest.WorkCreator do
     %{batch_size: Keyword.get(args, :batch_size, 20)}
   end
 
+  @spec create_works(any) :: {:noreply, any}
   @doc """
   Turn a batch of pending work rows into works
   """
   def create_works(state) do
```

Comment:
I've got a refactoring suggestion if you'd indulge me 😄 . There's a lot going on in `create_works/1` that I think could benefit from a change like this:

```suggestion
  def create_works(state) do
    with_log_level(:info, fn ->
      state
      |> get_and_update_pending_work_rows()
      |> handle_result()
    end)

    {:noreply, state}
  end

  defp get_and_update_pending_work_rows(state) do
    Repo.transaction(fn ->
      Progress.get_and_lock_pending_work_entries(state.batch_size)
      |> Progress.update_entries("CreateWork", "processing")
    end)
  end

  defp handle_result({:ok, []}), do: :noop

  defp handle_result({:ok, progress_rows}) do
    Logger.enable(self())
    Logger.info("Creating #{length(progress_rows)} works")
    create_works_from_rows(progress_rows |> Repo.preload(row: :sheet))
  end

  defp handle_result({:error, message}),
    do: Logger.error("Problem processing work entries: #{message}")
```

The test still pass, but I threw the logging error in the last `handle_result/1` without a ton of consideration. What do you think?

Reply:
I think this is a good refactoring, only I would return `[]` from the last `handle_result/1` just to respect the contract.